src/millie.nim

```diff
@@ -10,46 +10,41 @@ const units = @[
   "P", # quadrillion
 ]
 
-## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
+
 proc millify*(
     value: int64,
     precision: int = 1,
     lowercase: bool = false,
     space: bool = false,
   ): string =
+  ## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
   
-  # validate if `vaue` can be parsed or not
-  var val: SomeNumber = parseNum(value)
+  # preallocate the resulting string
+  result = newStringOfCap(10)
```

Comment:
Just for less memory allocations we can preallocate the string with the length we think is kind of average for the task